packages/fdc3-agent-proxy/src/channels/DefaultChannelSupport.ts

```diff
@@ -56,30 +64,42 @@ export class DefaultChannelSupport implements ChannelSupport {
       payload: {},
     };
     const response = await this.messaging.exchange<GetCurrentChannelResponse>(request, 'getCurrentChannelResponse');
-
-    if (response.payload?.channel?.id) {
+    //handle successful responses - errors will already have been thrown by exchange above
+    if (response.payload.channel) {
       return new DefaultChannel(
         this.messaging,
         response.payload.channel.id,
         'user',
         response.payload.channel.displayMetadata
       );
+    } else if (response.payload.channel === null) {
+      //this is a valid response if no channel is set
+      return null;
     } else {
+      console.warn(
+        'Invalid response from Desktop Agent to getCurrentChannel (channel should be explicitly null if no channel is set)!',
+        response.payload
+      );
       return null;
     }
   }
 
   async getUserChannels(): Promise<Channel[]> {
-    if (!this.userChannels) {
-      const request: GetUserChannelsRequest = {
+    const response = await this.messaging.exchange<GetUserChannelsResponse>(
```

Comment:
Is it possible to have a `const request: GetUserChannelsRequest`, for better type safety?

Reply:
Absolutely, probably another reversion, glad you're looking for them.